plutus-conformance/test-cases/uplc/evaluation/builtin/semantics/writeBits/case-31/case-31.uplc

```diff
@@ -1,13 +1,20 @@
--- Later occurences of duplicate indices take precedence.
+-- Later updates to duplicate indices take precedence.
 (program 1.0.0
  [
   [
+   (builtin writeBits)
    [
-    (builtin writeBits)
-    (con bytestring #00000000000000000000000000000000000000000000000000000000000000000000000000000000000000)
+    [
+     [
+      (builtin writeBits)
+      (con bytestring #00000000000000000000000000000000000000000000000000000000000000000000000000000000000000)
+     ]
+     (con (list integer) [340, 342, 343, 340, 342, 343])
+    ]
+    (con bool True)
    ]
-   (con (list integer) [340, 342, 343, 340, 342, 343])
+   (con (list integer) [340, 342, 343])
   ]
-  (con (list bool) [True, True, True, False, False, False])
+  (con bool False)
```

Comment:
Looking at this test I feel like maybe it's not the worst idea to have both versions of `writeBits`. Splitting the bits manually is error-prone and kinda annoying, plus it hurts performance when the list is known statically. We can of course express the old `writeBits` in terms of the new one, but this is going to be less efficient than just having the old one (in addition to the new one) and efficiency is what we care about here.

Reply:
In any scenario where the list is known statically (ie are constants) you should be able to automatically split the bits (precompilation) at zero onchain cost.